ecc/bls12-381/bandersnatch/point.go

```diff
@@ -168,6 +168,50 @@ func (p *PointAffine) IsOnCurve() bool {
 	return lhs.Equal(&rhs)
 }
 
+// IsInSubGroup checks if a point is in the prime subgroup
+// based on https://eprint.iacr.org/2022/037.pdf by D. Koshelev.
+func (p *PointAffine) IsInSubGroup() bool {
+	if p.IsZero() {
+		return true
+	}
+	initOnce.Do(initCurveParams)
+
+	// Given (x_e, x_e) a point on the twisted Edwards curve Ed_{a,d},
+	// (x_w, y_w) is a point on the birationally equivalent short Weierstrass curve W,
+	// where:
+	// 		x_w = ((1+y_e)/(1-y_e) + A/3 ) / B and
+	//  	A = 2(a+d)/(a-d), B = 4/(a-d)
+	//
+	// We need to check that the two tate pairings t_{2,P2}(P) and t_{2,Q2}(P) are 1,
```

Comment:
Should be renamed P_i (P1 and P2) instead of P2 and Q2, since you use the notation tate1, tate2 for the pairings.

Also, could you provide in the doc the x coordinates of P1 and P2 in the sw curve ?